activerecord/lib/active_record/persistence.rb

```diff
@@ -285,7 +285,7 @@ def instantiate(attributes, column_types = {}, &block)
       #
       # ==== Parameters
       #
-      # * +id+ - This should be the id or an array of ids to be updated.
+      # * +id+ - This should be the id, an array of ids to be updated, or nothing or :all for all objects.
```

Comment:
I think the original idea here was rather to *not* provide a first argument to apply to all objects, instead of relying on the special `:all` value.

Perhaps this should be framed as so:

```suggestion
      # * +id+ - This should be the id or an array of ids to be updated.
      #   Optional argument, defaults to all records in the relation.
```

Reply:
thanks good idea, I don't think the `:all` value is supposed to be exposed